sphinx and doxygen documentation (#56)
rapyuta-robotics/RapyutaSimulationPlugins: docs/source/components.rst

Components and main C++ classes
===============================
Overview and main C++ classes for components. 
Please check detail explanation of those class and other classes in `C++ Documentation <doxygen_generated/html/index.html>`_

Drives
------

| Drives provides feature to control robot. 
| Custom drive class should be created as child class of *RobotVehicleMovementComponent*

**example of class relations**

.. image:: doxygen_generated/html/d9/d8d/class_u_differential_drive_component__coll__graph.png

**main C++ classes**

> - `URobotVehicleMovementComponent <doxygen_generated/html/d7/d01/class_u_robot_vehicle_movement_component.html>`_:  Base Robot vehicle movement class which is used as part of #ARobotVehicle. Move robot without considering physics and publish odometry. This supports following platform such as elevator.

Reviewer: ```suggestion
- `URobotVehicleMovementComponent <doxygen_generated/html/d7/d01/class_u_robot_vehicle_movement_component.html>`_:  Base Robot vehicle movement class which is used as part of #ARobotVehicle. Move robot without considering physics and publish odometry. This supports adapting the robot pose to floor complex surfaces, and following moving platforms such as elevators.
```

| Custom robot class should be created as child class of *RobotVehicle* class. 
| *RRRobotVehicleROSController* create ROS2Node and control all drive and sensor components in the *RobotVehicle* class.

**main C++ classes**

- `ARobotVehicle <doxygen_generated/html/d7/d80/class_a_robot_vehicle.html>`_: Base RobotVehicle class. Other robot class should inherit from this class. Has URobotVehicleMovementComponent and initialize sensors.
- `ARRRobotVehicleROSController <doxygen_generated/html/d6/d83/class_a_r_r_robot_vehicle_r_o_s_controller.html>`_: Base Robot ROS controller class. Other robot controller class should inherit from this class. This class owns ROS2Node and provide ROS2 interfaces to control robot such as Twist msg.
- `ATurtlebotBurger <doxygen_generated/html/de/d76/class_a_turtlebot_burger.html>`_: Example of ARobotVehicle
- `ATurtlebotROSController <doxygen_generated/html/dd/d8f/class_a_turtlebot_r_o_s_controller.html>`_: Example of ARRRobotVehicleROSController

**main Blueprint classes classes**

- Turtlebot3: example of robots.
    - BP_TurtlebotROSController.uasset: inherit from C++ ARRRobotVehicleROSController. Example of construct robot controller with Blueprint.  

>     - Models: Static and skeltal meshes and materials.

Reviewer: ```suggestion
    - Models: Static and skeletal meshes and materials.
```

Core
----

| Core has base classes which are directly or indirectly used by other components in RapyutaSimulationPlugins Plugins.
| Core also has util classes as well. 

**main C++ classes(todo)**

- `ARRBaseActor <doxygen_generated/html/d9/d3d/class_a_r_r_base_actor.html>`_: Base actor class for all Rapyuta Sim actors
- `URRActorCommon <doxygen_generated/html/df/d29/class_u_r_r_actor_common.html>`_: todo
- Game objects
    - `ARRGameMode <doxygen_generated/html/d1/dbb/class_a_r_r_game_mode.html>`_: GameMode with specific setting, asset loading and ROS2 interface via ClockPublisher and ASimulationState.
    - `ARRGameState <doxygen_generated/html/d5/d01/class_a_r_r_game_state.html>`_: todo
    - `URRGameInstance <doxygen_generated/html/d1/d8e/class_u_r_r_game_instance.html>`_: This is a globally accessible instanced UObject that can store run-time data to be commonly accessed between levels and Scene instances.
    - `URRGameSingleton <doxygen_generated/html/d8/def/class_u_r_r_game_singleton.html>`_: todo
- Mesh
    - `ARRMeshActor <doxygen_generated/html/dd/de7/class_a_r_r_mesh_actor.html>`_: todo

>     - `URRProceduralMeshComponent <doxygen_generated/html/dc/d1a/class_u_r_r_procedural_mesh_component.html>`_: rocedural mesh components. this class is used to spawn robot and object from ROS2 service.

Reviewer: ```suggestion
    - `URRProceduralMeshComponent <doxygen_generated/html/dc/d1a/class_u_r_r_procedural_mesh_component.html>`_: procedural mesh components. this class is used to spawn robot and object from ROS2 service.
```

- Scene
    - `ARRSceneDirector <doxygen_generated/html/d6/d2f/class_a_r_r_scene_director.html>`_: todo
    - `URRSceneInstance <doxygen_generated/html/d7/d37/class_u_r_r_scene_instance.html>`_: todo
- Utils
    - `URRCoreUtils <doxygen_generated/html/de/db6/class_u_r_r_core_utils.html>`_: todo
    - `URRConversionUtils <doxygen_generated/html/d4/dc1/class_u_r_r_conversion_utils.html>`_: Data Conversion between ROS and UE. m <-> cm and Left handed <-> Right handed.

>     - `URRGeneralUtils <doxygen_generated/html/d5/d98/class_u_r_r_general_utils.html>`_: Other utils which is not belongs to other utils.

Reviewer: ```suggestion
    - `URRGeneralUtils <doxygen_generated/html/d5/d98/class_u_r_r_general_utils.html>`_: Other utils which do not belong to other utils.
```
